spec/unit/pops/types/error_spec.rb

```diff
@@ -0,0 +1,191 @@
+            notice($o)
+            notice(type($o))
+            CODE
+            expect(eval_and_collect_notices(code)).to eq([
+              "Error({'message' => 'bad tings happened', 'kind' => 'puppet/error'})",
+              "Error['puppet/error', 'ERROR']"
+            ])
+          end
+
+          it 'can be created from a hash' do
+            code = <<-CODE
+            $o = Error(message => 'Sorry, not implemented', kind => 'puppet/error', issue_code => 'NOT_IMPLEMENTED')
+            notice($o)
+            notice(type($o))
+            CODE
+            expect(eval_and_collect_notices(code)).to eq([
+              "Error({'message' => 'Sorry, not implemented', 'kind' => 'puppet/error', 'issue_code' => 'NOT_IMPLEMENTED'})",
+              "Error['puppet/error', 'NOT_IMPLEMENTED']"
+            ])
+          end
+
+          it 'is an instance of its type' do
+            code = <<-CODE
+            $o = Error('bad tings happened', 'puppet/error')
+            notice($o =~ type($o))
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['true'])
+          end
+
+          it 'is an instance of matching kind' do
+            code = <<-CODE
+            $o = Error('bad tings happened', 'puppet/error')
+            notice($o =~ Error[/puppet\\/error/])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['true'])
+          end
+
+          it 'is an instance of matching default issue' do
+            code = <<-CODE
+            $o = Error('bad tings happened')
+            notice($o =~ Error[default, 'ERROR'])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['true'])
+          end
+
+          it 'is an instance of matching issue' do
+            code = <<-CODE
+            $o = Error('bad tings happened', 'puppet/error', 'FEE')
+            notice($o =~ Error[default, Enum['FOO', 'FEE', 'FUM']])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['true'])
+          end
+
+          it 'is an instance of matching kind and issue' do
+            code = <<-CODE
+            $o = Error('bad tings happened', 'puppet/error', 'FEE')
+            notice($o =~ Error['puppet/error', Enum['FOO', 'FEE', 'FUM']])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['true'])
+          end
+
+          it 'is not an instance unless kind matches' do
+            code = <<-CODE
+            $o = Error('bad tings happened', 'puppetlabs/error')
+            notice($o =~ Error[/puppet\\/error/])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['false'])
+          end
+
+          it 'is not an instance if default issue is not matched' do
+            code = <<-CODE
+            $o = Error('bad tings happened', undef)
+            notice($o =~ Error[default, 'OTHER'])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['false'])
```

Comment:
I don't understand this test - the title is strange and it tests two things not matching at the same time

Reply:
+ 'ERROR' issue_code as default...